sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1459,7 +1459,7 @@ DATA: 'DATA';
 DATABASE: 'DATABASE';
 DATABASES: 'DATABASES' | 'SCHEMAS';
 DAY: 'DAY';
-DAYS: 'DAYS';
+DAYS: 'DAYS'| 'D';
```

Comment:
`D` is not even a keyword in pgsql: https://www.postgresql.org/docs/8.1/sql-keywords-appendix.html

I'm curious how pgsql supports these abbreviations.

Reply:
yes, not abbreviations but also its plural

Reply:
It seems that in pg the type_name interval is required

Reply:
```sql
postgres=# select  '1 s 2 days 6 weeks 3 seconds';
           ?column?
------------------------------
 1 s 2 days 6 weeks 3 seconds
(1 row)

postgres=# select  '1 s 2 days 6 weeks';
      ?column?
--------------------
 1 s 2 days 6 weeks
(1 row)

postgres=# select  interval '1 s 2 days 6 weeks';
     interval
------------------
 44 days 00:00:01
(1 row)

postgres=# select  interval '1 s 2 days 3 weeks';
     interval
------------------
 23 days 00:00:01
(1 row)

postgres=# select  interval '1 s 2 days d weeks';
    interval
-----------------
 2 days 00:00:01
(1 row)

postgres=# select  interval '1 s 2 days days weeks';
    interval
-----------------
 2 days 00:00:01
(1 row)

postgres=# select  interval '1 s 2 days year weeks';
    interval
-----------------
 2 days 00:00:01
(1 row)

postgres=# select  interval '1 s 2 yrs days weeks';
     interval
------------------
 2 years 00:00:01
(1 row)
```

Reply:
proposed a new approach to parse intervals in the form of abbreviations, without adding new keywords to parser, please take a look, thanks.